config/config.go

```diff
@@ -2,25 +2,51 @@ package config
 
 import (
 	"fmt"
-	"os"
 	"strings"
 	"syscall"
 
-	"github.com/BurntSushi/toml"
+	"github.com/ipfs/go-log"
 	"github.com/keep-network/keep-common/pkg/chain/ethereum"
+	ethereumCommon "github.com/keep-network/keep-common/pkg/chain/ethereum"
+	"github.com/keep-network/keep-core/pkg/beacon/registry"
+	ethereumChain "github.com/keep-network/keep-core/pkg/chain/ethereum"
+	ethereumBeaconGen "github.com/keep-network/keep-core/pkg/chain/ethereum/beacon/gen"
 	"github.com/keep-network/keep-core/pkg/net/libp2p"
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 	"golang.org/x/crypto/ssh/terminal"
 )
 
-// #nosec G101 (look for hardcoded credentials)
-// This line doesn't contain any credentials.
-// It's just the name of the environment variable.
-const passwordEnvVariable = "KEEP_ETHEREUM_PASSWORD"
+var logger = log.Logger("keep-config")
+
+// Environment variables names for properties configuration. For simplification we
+// bind the names that are different from the auto-generated by viper with the
+// `viper.AutomaticEnv()` function. For example instead of expecting
+// `KEEP_ETHEREUM_ACCOUNT_KEYFILEPASSWORD` we expect `KEEP_ETHEREUM_PASSWORD`.
+const (
+	// Environment Variables
+	// #nosec G101 (look for hardcoded credentials)
+	// This line doesn't contain any credentials.
+	// It's just the name of the environment variable.
+	ethereumPasswordEnvVariable = "KEEP_ETHEREUM_PASSWORD"
+
+	configFilePathFlag = "config"
+
+	// Command Line Flags
+	dataDirFlag                 = "datadir"
+	ethereumKeyFlag             = "ethereum.key"
+	ethereumUrlFlag             = "ethereum.url"
+	ethereumContractsPrefixFlag = "ethereum.contracts"
+	portFlag                    = "network.port"
+	peersFlag                   = "network.peers"
+	announcedAddressesFlag      = "network.announced-addresses"
```

Comment:
This is way better now but I think we should improve it. In particular:
- Parameter name for the storage is `datadir` but in the configuration file it's specified under `[storage] DataDir`. Mapping between these two values is not straightforward. Same about `ethereum.key` parameter and `[ethereum.account] KeyFile`.
- Some parameters can be specified in CLI and some others can not. We should go for "all or nothing" option, otherwise, this is confusing. Specifically, being able to configure metrics and diagnostics ports for Kubernetes is also important.

I would also enable metrics and diagnostics by default. Most node operators for v1 had no idea they existed. 

Let's do it now because every change later will require updates to each developer's local config, our testnet config, and the config of operators integrating with our testnet.

Here is my proposition how the keep client `--help` should look like:

```

ETHEREUM OPTIONS:
  --ethereum.url                          WS connection URL for Ethereum client.
  --ethereum.keyFile                      The local filesystem path to Keep operator account keyfile.
  --ethereum.miningCheckInterval          The time interval in seconds in which transaction mining status is checked. If the transaction is not mined within this time, the gas price is increased and transaction is resubmitted. (default: 60)
  --ethereum.maxGasFeeCap                 The maximum gas fee the client is willing to pay for the transaction to be mined. If reached, no resubmission attempts are performed. (default: 500 Gwei)
  --ethereum.requestPerSecondLimit        Request per second limit for all types of Ethereum client requests. (default: 150)
  --ethereum.concurrencyLimit             The maximum number of concurrent requests which can be executed against Ethereum client. (default: 30)
  --ethereum.balanceAlertThreshold        The minimum balance of operator account below which client starts reporting errors in logs. (default: 0.5 ether)


NETWORK OPTIONS:
  --network.peers                         Addresses of the network bootstrap nodes.
  --network.port                          Keep client listening port.
  --network.announcedAddresses            Overwrites the default Keep client address announced in the network. Should be used for NAT or when more advanced firewall rules are applied.
  --network.disseminationTime             Specifies courtesy message dissemination time in seconds for topics the node is not subscribed to. Should be used only on selected bootstrap nodes. (default: none)


STORAGE OPTIONS:
  --storage.dataDir                       Location to store the Keep client key shares and other sensitive data.


METRICS OPTIONS:
  --metrics.port                          Metrics HTTP server listening port. (default: 8080)
  --metrics.networkMetricsTick            Network metrics check tick in seconds. (default: 60)
  --metrics.ethereumMetricsTick           Ethereum metrics check tick in seconds. (default: 600)


DIAGNOSTICS OPTIONS:
  --diagnostics.port                      Diagnostics HTTP server listening port. (default: 8081)


DEVELOPER OPTIONS:
  --developer.randomBeaconAddress         Address of RandomBeacon smart contract
  --developer.tokenStakingAddress         Address of TokenStaking smart contract
  --developer.walletRegistryAddress       Address of WalletRegistry smart contract
```

And here is how we should update the config file structure:
```
[ethereum]
	URL                = "ws://127.0.0.1:8546"
	KeyFile            = "/Users/someuser/ethereum/data/keystore/UTC--2018-03-11T01-37-33.202765887Z--AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8AAAAAAAAA"

	# Uncomment to override the defaults for transaction status monitoring.
	#
	# MiningCheckInterval is the interval in which transaction
	# mining status is checked. If the transaction is not mined within this
	# time, the gas price is increased and transaction is resubmitted.
	#
	# MiningCheckInterval = 60  # 60 sec (default value)
	#
	#
	# MaxGasFeeCap specifies the maximum gas fee cap the client is
	# willing to pay for the transaction to be mined. The offered transaction
	# gas cost can not be higher than the max gas fee cap value. If the maximum
	# allowed gas fee cap is reached, no further resubmission attempts are
	# performed. This property should be set only for Ethereum. In case of
	# legacy non-EIP-1559 transactions, this field works in the same way as
	# `MaxGasPrice` property.
	#
	# MaxGasFeeCap = "500 Gwei" # 500 Gwei (default value)
	
	# Uncomment to enable Ethereum node rate limiting. Both properties can be
	# used together or separately.
	#
	# RequestsPerSecondLimit sets the maximum average number of requests
	# per second which can be executed against the Ethereum node.
	# All types of Ethereum node requests are rate-limited,
	# including view function calls.
	#
	# RequestsPerSecondLimit = 150
	#
	# ConcurrencyLimit sets the maximum number of concurrent requests which
	# can be executed against the Ethereum node at the same time.
	# This limit affects all types of Ethereum node requests,
	# including view function calls.
	#
	# ConcurrencyLimit = 30
	
	# BalanceAlertThreshold defines a minimum value of the operator's account
	# balance below which the client will start reporting errors in logs.
	# A value can be provided in `wei`, `Gwei` or `ether`, e.g. `7.5 ether`,
	# `7500000000 Gwei`.
	#
	# BalanceAlertThreshold = "0.5 ether" # 0.5 ether (default value)

[network]
 	Peers = ["/ip4/127.0.0.1/tcp/3919/ipfs/16Uiu2HAmFRJtCWfdXhZEZHWb4tUpH1QMMgzH1oiamCfUuK6NgqWX"]
 	Port = 3920
	
	# Uncomment to override the node's default addresses announced in the network
	# AnnouncedAddresses = ["/dns4/example.com/tcp/3919", "/ip4/80.70.60.50/tcp/3919"]
	
	# Uncomment to enable courtesy message dissemination for topics this node is
	# not subscribed to. Messages will be forwarded to peers for the duration
	# specified as a value in seconds.
	# Message dissemination is disabled by default and should be enabled only
	# on selected bootstrap nodes. It is not a good idea to enable dissemination
	# on non-bootstrap node as it may clutter communication and eventually lead
	# to blacklisting the node. The maximum allowed value is 90 seconds.
	#
	# DisseminationTime = 90

[storage]
  DataDir = "/my/secure/location"

# Metrics collects and exposes information useful for external monitoring tools usually
# operating on time series data.
# All values exposed by metrics module are quantifiable or countable.
#
# The following metrics are available:
# - connected peers count
# - connected bootstraps count
# - eth client connectivity status
[Metrics]
    Port = 8080
    NetworkMetricsTick = 60
    EthereumMetricsTick = 600

# Diagnostics module exposes the following information:
# - list of connected peers along with their network id and ethereum operator address
# - information about the client's network id and ethereum operator address
[Diagnostics]
    Port = 8081

# Developer options to work with locally deployed contracts
#
# [developer]
#	TokenStakingAddress = "0xBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB"
#	RandomBeaconAddress = "0xBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB"
#	WalletRegistryAddress = "0xBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB"
```